gv.c

```diff
@@ -2600,9 +2610,27 @@ Perl_gv_fetchpvn_flags(pTHX_ const char *nambeg, STRLEN full_len, I32 flags,
     if (!stash && !find_default_stash(&stash, name, len, is_utf8, add, sv_type)) {
         return NULL;
     }
+    char tmp[30];
+    char *use_name = (char*)name;
+    STRLEN use_len = len;
+
+    if (((U8)name[0]) < 32) {
+        use_name = S_is_internal_caret_name(use_name, &use_len, tmp);
```

Comment:
Why doesn't S_is_internal_caret_name just return a pointer to a static string?

Reply:
As in a global static var? Probably because I just simply didn't think of it. :-)  I suspect I was influenced by the fact that not all names get translated, and this "works". IOW: no good reason.